File: backend/app/schemas/rating.py
"""Pydantic schemas for course ratings."""

import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class RatingCreate(BaseModel):
    rating: int = Field(..., ge=1, le=5)
    review: str | None = Field(None, max_length=2000)


class RatingUpdate(BaseModel):
    rating: int | None = Field(None, ge=1, le=5)
    review: str | None = Field(None, max_length=2000)


class RatingResponse(BaseModel):
    id: uuid.UUID
    user_id: uuid.UUID
    course_id: uuid.UUID
    rating: int
    review: str | None = None
    user_name: str = ""
    user_avatar: str | None = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class RatingListResponse(BaseModel):
    items: list[RatingResponse]
    total: int
    average: float = 0.0

File: frontend/src/types/rating.ts
/**
 * TypeScript types for course ratings and reviews.
 */
export interface Rating {
  id: string;
  user_id: string;
  course_id: string;
  rating: number;
  review: string | null;
  user_name: string;
  user_avatar: string | null;
  created_at: string;
  updated_at: string;
}

export interface RatingListResponse {
  items: Rating[];
  total: number;
  average: number;
}

export interface RatingCreate {
  rating: number;
  review?: string | null;
}

export interface RatingUpdate {
  rating?: number;
  review?: string | null;
}
